server/topic.go

```diff
@@ -3342,29 +3346,44 @@ func (t *Topic) subsCount() int {
 
 // Adds a new multiplex proxied session to the topic's clusterWriteLoop.
 func (t *Topic) addProxiedSession(s *Session) {
+	// Send an interrupt signal to clusterWriteLoop that a new session
+	// is being added and acquire the lock.
+	if len(t.proxiedChannels) > 0 {
+		interruptChan := t.proxiedChannels[0].Chan.Interface().(chan struct{})
+		for !t.proxiedLock.TryLock() {
+			interruptChan <- struct{}{}
+		}
+	} else {
+		t.proxiedLock = concurrency.NewSimpleMutex()
+		t.proxiedLock.Lock()
+	}
+	// At this point we are guaranteed to have grabbed t.proxiedLock.
 	t.proxiedSessions = append(t.proxiedSessions, s)
 	if len(t.proxiedSessions) == 1 {
 		t.proxiedChannels = make([]reflect.SelectCase, 1+3)
 		continueChan := make(chan struct{})
 		t.proxiedChannels[0] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(continueChan)}
-		t.proxiedChannels[1] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(s.send)}
-		t.proxiedChannels[2] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(s.stop)}
-		t.proxiedChannels[3] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(s.detach)}
+		t.proxiedChannels[EventSend+1] = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(s.send)}
```

Comment:
Would if look cleaner if events are defined as `EventSend ProxyEventType = iota + 1` instead of `... = iota`?

Reply:
Or maybe literally = 1, = 2...

Reply:
Sure.